### ISSUE TITLE
DevServer return incorrect Content-Type for assets files

### ISSUE THREAD

**Issue by timwangdev** [callstack/haul#335]
### Current Behavior

`GET /assets/ios/index.html 200`
```
Accept-Ranges:bytes
Access-Control-Allow-Origin:*
Connection:keep-alive
Content-Type:application/javascript
X-Powered-By:Express
...
```

### Expected Behavior

Should return correct Content-Type based on file extension names. 

### Your Environment

| software         | version
| ---------------- | -------
| Haul             |  1.0.0-beta.12
| react-native     |  0.52.0
| node             | 8.9.1
| npm or yarn      | yarn 1.3.2

**Comment by timwangdev** [comment]
@joearasin

And related comment https://github.com/callstack/haul/issues/120#issuecomment-358226553

**Comment by Krizzu** [comment]
Can you tell more about your use case?

**Comment by timwangdev** [comment]
@Krizzu I was trying to use local html file for `WebView` 
using `<WebView source={require('./some.html')} />`.

On dev mode it displays raw html string on the view but works fine in production bundle.

After inspection I found all the assets files returned from haul dev server have incorrect content-type header as `application/javascript`. 

I'm not familiar with webpack dev server but I think there is some misconfiguration.

**Comment by satya164** [comment]
Webpack dev middleware has an option to set mimetypes: https://github.com/webpack/webpack-dev-middleware#mimetypes

We can add the mimetypes for all the extensions we support by default. PR welcome :)

**Comment by Krizzu** [comment]
@timwangdev What's expected from `require('./some.html')` ? If you receive `raw html`, you can pass it to the WebView via its source prop 

```js
const rawHtml = require('./some.html'); 

// ...
	<WebView source={{html: rawHtml}} />
```

**Comment by timwangdev** [comment]
@Krizzu Webpack assetLoader https://github.com/callstack/haul/blob/master/src/loaders/assetLoader.js will transform `.html` module to something like this.
```js
module.exports = AssetRegistry.registerAsset({ 
  __packager_asset: true,
  scales: [1],
  name: "local",
  type: "html",  
  hash: "1417faac1f362c5783cbe82b652a8bbf",
  httpServerLocation: "assets/ios/",
  fileSystemLocation: "/path/app/",
});
```
This one is valid and react-native will try to fetch this resource from local when rendering the screen.

`<WebView source={{html: rawHtml}} />` is also working by using [webpack-contrib/raw-loader](https://github.com/webpack-contrib/raw-loader), this loader will just import files as string into js bundle, but it's not good for reduce bundle size and optimize js parse time.

@satya164 I think that `mimetypes` option are intended for register custom extension, may not suitable for this case?

**Comment by alexanderkjeldaas** [comment]
When loading built javascript dynamically in Chrome, this bug ensures that the scripts are blocked by the CORB system in Chrome.
https://chromium.googlesource.com/chromium/src/+/master/services/network/cross_origin_read_blocking_explainer.md

**Comment by zamotany** [comment]
Closing due to inactivity. If you're having any similar problems, feel free to open a new issue and make sure you're using `@haul-bundler/cli`.
